yt_dlp/extractor/subsplash.py

```diff
@@ -0,0 +1,112 @@
+import functools
+import math
+
+from .common import InfoExtractor
+from ..utils import (
+    InAdvancePagedList,
+    int_or_none,
+    str_or_none,
+    traverse_obj,
+    unified_strdate,
+    url_or_none,
+)
+
+
+class SubsplashVideoIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?subsplash\.com/u/[^/]+/media/d/(?P<id>\w+)'
+    _TESTS = [{
+        'url': 'https://subsplash.com/u/skywatchtv/media/d/5whnx5s-the-grand-delusion-taking-place-right-now',
+        'md5': '2d67c50deac3c6c49c6e25c4a5b25afe',
+        'info_dict': {
+            'id': '33f8d305-68ab-414c-acf2-f2317a0abe21',
+            'ext': 'mp4',
+            'title': 'THE GRAND DELUSION TAKING PLACE RIGHT NOW!',
+            'description': 'md5:220a630865c3697b0ec9dcb3a70cbc33',
+            'upload_date': '20240901',
+            'duration': 1710,
+            'thumbnail': r're:^https?://.*\.(?:jpg|png)$',
```

Comment:
```suggestion
            'thumbnail': r're:https?://.*\.(?:jpg|png)$',
```
See https://discord.com/channels/807245652072857610/1112613156934668338/1293369656098160702. `re.match` is used here when running tests(which matches from the start of the string already)